### PR TITLE
PYTHON-3885 Use tox in remaining Evergreen Tests

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .evergreen/run-mongodb-aws-test.sh [hunk 1]
```diff
@@ -39,37 +39,12 @@ fi
...
-    fi
-    python -m pip install '.[aws]'
-    python test/auth_aws/test_auth_aws.py -v
-    deactivate
-    rm -rf venvaws
-}
-
-PYTHON=${PYTHON_BINARY:-}
-if [ -z "$PYTHON" ]; then
+if [ -z "$PYTHON_BINARY" ]; then
```
Comment (NoahStapp) [inline .evergreen/run-mongodb-aws-test.sh:42]: Do we need this file at all? What if we put the new contents directly into `config.yml`?

File: .evergreen/run-mongodb-oidc-test.sh [hunk 2]
```diff
@@ -45,41 +45,13 @@ if [ "$ASSERT_NO_URI_CREDS" = "true" ]; then
...
-    python -m pip install -U pip setuptools
-    python -m pip install '.[aws]'
-    python test/auth_aws/test_auth_oidc.py -v
-    deactivate
-    rm -rf venvoidc
-}
-
-PYTHON=${PYTHON_BINARY:-}
-if [ -z "$PYTHON" ]; then
+if [ -z "$PYTHON_BINARY" ]; then
```
Comment (NoahStapp) [inline .evergreen/run-mongodb-oidc-test.sh:48]: Same for this file

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .evergreen/run-mongodb-aws-test.sh (hunk 1 above)
Comment (blink1073) [reply .evergreen/run-mongodb-aws-test.sh:42]: We would still run these two files as entrypoints for local testing.  I wanted to keep the logic out of config.yml where possible, and there are parts of this that we don't want to echo and parts that we do.

Comment (NoahStapp) [reply .evergreen/run-mongodb-aws-test.sh:42]: Makes sense!

**Review by NoahStapp — APPROVED** [review]

**Comment by blink1073** [comment]
Note: the OIDC failure is tracked in [PYTHON-3909](https://jira.mongodb.org/browse/PYTHON-3909)

**Pull request merged**